lib/cli/drive-scanner.js

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2016 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const drivelist = require('drivelist');
+const EventEmitter = require('events').EventEmitter;
+const _ = require('lodash');
+const Rx = require('rx');
+
+const interval = new WeakMap();
+const prevPollDrives = new WeakMap();
```

Comment:
What is the reason to use maps if we only get/set from/to a single key (`this`)?